### PR TITLE
Codecov.io support

### PR DESCRIPTION
**Pull request by gundalow**
I'm concerned that test coverage isn't that good. Lets see if we can generate some metrics to prove this either way.

### CODE REVIEW

File: shippable.yml [hunk 1]
```diff
@@ -12,3 +15,11 @@ build:
    - shippable_retry pip install -Ur test-requirements.txt
 
    - pytest
+
+after_success:
+
+  - codecov --token=272fc7a8-bd55-4e0c-af2f-7d475d0f3c0a -r ansible/ansibullbot
```
Comment (webknjaz) [inline shippable.yml]: @gundalow it shouldn't require a token for public CIs, just `codecov` command should be enough, it uploads the report where needed.

File: shippable.yml [hunk 2]
```diff
@@ -12,3 +15,11 @@ build:
...
+
+  - coverage xml -o $COVERAGE_REPORTS/coverage.xml
```
Comment (webknjaz) [inline shippable.yml]: What are these supposed to do? pytest already puts a report in that folder. See `pytest.ini`.

File: test-requirements.txt [hunk 3]
```diff
@@ -5,6 +5,7 @@ pip>=9.0
 -r requirements.txt
 
 # Test runner:
+codecov
```
Comment (webknjaz) [inline test-requirements.txt]: It's not about runner, please put it into a separate section, like `# Reporting:`

**Comment by webknjaz** [comment]
@gundalow currently pytest-cov generates coverage and Shippable even collects it from there, so in each job it's available in a separate tab, like https://app.shippable.com/github/ansible/ansibullbot/runs/731/2/coverage

you'll only need to specify `codecov -f shippable/codecoverage/pytest-cov.xml -X gcov`

Examples:
* https://github.com/cherrypy/cheroot/blob/master/tox.ini#L16 — how I do it in another project
* https://github.com/cherrypy/cheroot/blob/master/.codecov.yml — how you can configure codecov slightly differently from defaults

**Comment by webknjaz** [comment]
@gundalow I've adjusted everything. It's ready to merge.

**Comment by gundalow** [comment]
@webknjaz Thanks!

**Pull request merged**